models/lpjguess/R/write.config.LPJGUESS.R

```diff
@@ -80,43 +81,80 @@ write.config.LPJGUESS <- function(defaults, trait.values, settings, run.id){
-#       }
-#     }
-#     if (filename == "") {
-#       logger.severe("Could not find config template")
-#     }
-#     logger.info("Using", filename, "as template")
-#     config.text <- readLines(con=filename, n=-1)
-#   }
-#   
-#   config.text <- gsub('@SITE_LAT@', settings$run$site$lat, config.text)
-#   config.text <- gsub('@SITE_LON@', settings$run$site$lon, config.text)
-#   config.text <- gsub('@SITE_MET@', settings$run$inputs$met$path, config.text)
-#   config.text <- gsub('@MET_START@', settings$run$site$met.start, config.text)
-#   config.text <- gsub('@MET_END@', settings$run$site$met.end, config.text)
-#   config.text <- gsub('@START_MONTH@', format(startdate, "%m"), config.text)
-#   config.text <- gsub('@START_DAY@', format(startdate, "%d"), config.text)
-#   config.text <- gsub('@START_YEAR@', format(startdate, "%Y"), config.text)
-#   config.text <- gsub('@END_MONTH@', format(enddate, "%m"), config.text)
-#   config.text <- gsub('@END_DAY@', format(enddate, "%d"), config.text)
-#   config.text <- gsub('@END_YEAR@', format(enddate, "%Y"), config.text)
-#   config.text <- gsub('@OUTDIR@', settings$host$outdir, config.text)
-#   config.text <- gsub('@ENSNAME@', run.id, config.text)
-#   config.text <- gsub('@OUTFILE@', paste('out', run.id, sep=''), config.text)
-#  
-#   #-----------------------------------------------------------------------
-#   config.file.name <- paste0('CONFIG.',run.id, ".txt")
-#   writeLines(config.text, con = paste(outdir, config.file.name, sep=''))
+}
+
+#==================================================================================================#
+#' @name write.insfile.LPJGUESS
+#' @title Write LPJ-GUESS instruction script
+#' @export
+#' @param settings PEcAn settings list
+#' @param run.id PEcAn run ID
+#' @return settings Updated list
+#' @author Istem Fer
+
+write.insfile.LPJGUESS <- function(settings, trait.values, rundir, outdir, run.id){
+  
+  guessins <- readLines(con=system.file("template.ins", package = "PEcAn.LPJGUESS"), n=-1)
+  paramsins <- readLines(con=system.file("pecan.ins", package = "PEcAn.LPJGUESS"), n=-1)
+  
+  # write parameter values
+  param.names <- lapply(seq_along(settings$pfts), function(x) paste0(names(trait.values)[x], "_", names(trait.values[[x]]), ".*"))
+
+  # write params with values from trait.values  
+  for(i in seq_along(settings$pfts)){
+    for(n in seq_along(trait.values[[i]])){
+      paramsins <- gsub(param.names[[i]][n], trait.values[[i]][n], paramsins)
+    }
+  }
+  
+  # if anything is not replaced use the defaults param value
+  # if ".@" exists, remove the string infront of ".@" and use the value 
+  for(i in seq_along(paramsins)){
+    if(grepl(".@",paramsins[i])){
+      paramsins[i] <- gsub("\\s*\\w*$", paste0(" ", gsub("^[^:]*.@", "", paramsins[i])), gsub(".@.*","", paramsins[i]))
+    }
+  }
+  
+  
+  
+  # write clim file names
+
+  tmp.file <- settings$run$inputs$met$path
```

Comment:
I don't exactly remember what was being passed as met$path after METProcess, it was the path to the temperature I believe, but if not it should be an easy fix